### ISSUE TITLE
(defconstant x 10) (defun foo (x))   breaks down

### ISSUE THREAD

**Issue by drmeister** [drmeister/clasp#201]

**Comment by Shinmera** [comment]
Well, on b8c6690535bdc61bfdaeb50dbe23cd89633200a5
```
> (defconstant x 10) (defun foo (x))

X
> 
Condition of type: SIMPLE-PROGRAM-ERROR
Unknown initialization option :CODE for class #<STANDARD-CLASS CLEAVIR-CODE-UTILITIES:REQUIRED-MUST-BE-VARIABLE>
```
So it seems to try to error right but fails due to a different bug. Might be an issue in Cleavir itself, I can't tell. @robert-strandh

**Comment by robert-strandh** [comment]
Hello,

I am pretty sure I fixed this a few months ago.  Maybe Clasp is using an
old version.

On Fri, Mar 4, 2016 at 8:49 AM, Nicolas Hafner <notifications@github.com>
wrote:

> Well, on b8c6690
> <https://github.com/drmeister/clasp/commit/b8c6690535bdc61bfdaeb50dbe23cd89633200a5>
>
> > (defconstant x 10) (defun foo (x))
>
> X
> >
> Condition of type: SIMPLE-PROGRAM-ERROR
> Unknown initialization option :CODE for class #<STANDARD-CLASS CLEAVIR-CODE-UTILITIES:REQUIRED-MUST-BE-VARIABLE>
>
> So it seems to try to error right but fails due to a different bug. Might
> be an issue in Cleavir itself, I can't tell. @robert-strandh
> <https://github.com/robert-strandh>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/drmeister/clasp/issues/201#issuecomment-192169611>.
>



-- 
Robert Strandh

**Comment by Shinmera** [comment]
Fixed on 99669d674a6f2b7ce0a9127cf6537fc03066baa5
```
> (defconstant x 10) (defun foo (x))

X
> 
Condition of type: REQUIRED-MUST-BE-VARIABLE
In this type of lambda list, the required parameter must
be a variable which is also not the name of a constant.
But the following was found instead:
X

```
